File: examples/temperature-listener.py
import sys
from subprocess import Popen, PIPE

# Store temperatures.
temperatures = []
sensor = Popen(['node', 'sensor.js'], stdout=PIPE)
buffer = b''
while True:

    # Read sensor data one char at a time
    out = sensor.stdout.read(1)

    # After a full reading...
    if out == b'\n':
        temperatures.append(float (buffer))
        print(temperatures)
        buffer = b''
    else:
        # Append to the buffer.
        buffer += out

File: examples/temperature-listener.js
const { spawn } = require('child_process');
// Storage for the readings.
const temperatures = [];

const sensor = spawn('spawn', ['sensor.py']);
sensor.stdout.on('data', function(data) {
    // Coerce Buffer object to Float
    temperatures.push(parseFloat(data));

    // Log to debug
    console.log(temperatures);
});